lib/commands/actions.js

```diff
@@ -1,23 +1,231 @@
 import { fs, tempDir } from 'appium-support';
 import path from 'path';
-import { addMedia } from 'node-simctl';
+import log from '../logger';
+import { exec } from 'teen_process';
+import { addMedia, getAppContainer } from 'node-simctl';
+
+const CONTAINER_PATH_MARKER = '@';
 
 let commands = {};
 
-commands.pushFile = async function (remotePath, base64Data) {
-  if (!this.isSimulator()) {
-    throw new Error('Media upload is only supported on Simulator');
+async function verifyIFusePresence () {
+  if (!await fs.which('ifuse')) {
+    log.errorAndThrow(`'ifuse' tool is required to be installed on the machine. ` +
+                      `Install it using 'brew cask install osxfuse && brew install ifuse' or check ` +
+                      `if it is available in PATH environment variable if the tool is already installed. ` +
+                      `Current PATH value: ${process.env.PATH}`);
+  }
+}
+
+async function mountDevice (device, iFuseArgs) {
+  log.debug(`Starting ifuse with args '${iFuseArgs}'...`);
+  try {
+    await exec('ifuse', iFuseArgs);
+  } catch (e) {
+    log.errorAndThrow(`Cannot mount the media folder of the device with UDID ${device.udid}. ` +
+                      `Make sure osxfuse plugin has necessary permissions in System Preferences->Security & Privacy. ` +
+                      `Error code: ${e.code}; stderr output: ${e.stderr}`);
+  }
+}
+
+/**
+ * Save the given base64 data chunk as a binary file on the Simulator under test.
+ *
+ * @param {Object} device - The device object, which represents the device under test.
+ *                          This object is expected to have the `udid` property containing the
+ *                          valid device ID.
+ * @param {string} remotePath - The remote path on the device. This variable can be prefixed with
+ *                              bundle id, so then the file will be uploaded to the corresponding
+ *                              application container instead of the default media folder, for example
+ *                              '@com.myapp.bla/RelativePathInContainer/111.png'. The '@' character at the
+ *                              beginning of the argument is mandatory in such case.
+ *                              The relative folder path is ignored if the file is going to be uploaded
+ *                              to the default media folder and only the file name is considered important.
+ * @param {string} base64Data - Base-64 encoded content of the file to be uploaded.
+ */
+async function pushFileToSimulator (device, remotePath, base64Data) {
+  if (remotePath.startsWith(CONTAINER_PATH_MARKER)) {
+    if (!remotePath.includes('/')) {
+      log.errorAndThrow(`It is expected that bundle identifier is separated from the relative path with a single slash. ` +
+                        `'${remotePath}' is given instead`);
+    }
+    const bundleId = remotePath.substring(CONTAINER_PATH_MARKER.length, remotePath.indexOf('/'));
```

Comment:
All of this seems built for a regular expression 
```js
let remotePath = '@com.myapp.bla/RelativePathInContainer/111.png';
if (if (remotePath.startsWith(CONTAINER_PATH_MARKER)) {
  let match = /@([^\/]+)\/(.+)/.exec(remotePath)
  if (match) {
    let bundleId = match[1];
    let relativePath = match[2];
    // ...
  } else {
    // throw error
  }
}
```

Reply:
fixed